src/Microsoft.Identity.Client/Internal/Cache/TokenCacheKey.cs

```diff
@@ -137,7 +138,7 @@ public override int GetHashCode()
             const string Delimiter = ":::";
             return (this.Authority + Delimiter
                     + MsalHelpers.AsSingleString(this.Scope) + Delimiter
-                    + this.ClientId.ToLower() + Delimiter
+                    + this.ClientId.ToLowerInvariant() + Delimiter
```

Comment:
By default ToLower() uses the CurrentCulture, which is probably not what we want here.